client/components/HeaderNav.vue

```diff
@@ -114,21 +114,42 @@
             </HeadlessMenuItems>
           </transition>
         </HeadlessMenu>
-        <div v-else class="-m-1.5 flex items-center p-1.5">
-          <a href="/oidc/authenticate/" class="hidden lg:flex items-center text-sm dark:text-neutral-300">
-            <span class="font-semibold">Login</span>
-            <Icon name="solar:alt-arrow-right-line-duotone" size="1.5em" class="ml-1" />
-          </a>
-        </div>
+        <DialogRoot :open="!userStore.authenticated">
+          <DialogPortal>
+            <DialogOverlay class="backdrop-blur-sm z-[100] bg-black/30 fixed inset-0" />
```

Comment:
The `z-[100]` could, I think, be `z-100` without affecting anything. Is there a reason to prefer the brackets, or is it just consistency?

Reply:
Brackets are usually for custom values but z-index handles all integers correctly without it as well. Both are valid so I would leave it to Matthew's preference.

Reply:
My take on it is that in general I think square bracket custom styles should be discouraged (or at least viewed with suspicion!) for CSS filesize reasons / consolidating unnecessary styles. They are often bespoke styles that aren't necessary, like using a custom box shadow vs a plain `shadow-xl` and the later should be preferred.

And arguably they should all be refactored into our own `Dialog.vue` file for reuse across the project, but I wasn't sure about commonalities yet.

The `z-*` styles will probably need rationalizing later.